File: C++/AbstractFactory/WidgetFactory/scrollbar.cc
#include <iostream>
#include "scrollbar.h"

void WindowsStyleScrollbar::scroll() const {
    std::cout << "Scrolling this Windows-style scrollbar!!!\n";
}


void MacStyleScrollbar::scroll() const {
    std::cout << "Scrolling this Mac-style scrollbar!!!\n";
}

File: Java/AbstractFactory/WidgetFactory/src/com/ericgtkb/Scrollbar.java
package com.ericgtkb;

public interface Scrollbar {
    void scroll();
}

class WindowsStyleScrollbar implements Scrollbar {
    @Override
    public void scroll() {
        System.out.println("Scrolling this Windows-style scrollbar!!!");
    }
}

class MacStyleScrollbar implements Scrollbar {
    @Override
    public void scroll() {
        System.out.println("Scrolling this Mac-style scrollbar!!!");
    }
}
